core/lib/src/config/address.rs

```diff
@@ -0,0 +1,63 @@
+use std::{io, fmt};
+use std::str::FromStr;
+#[cfg(unix)] use std::path::PathBuf;
+use std::net::{IpAddr, ToSocketAddrs};
+
+/// An enum corresponding to the address to serve on
+#[derive(Debug, Clone, PartialEq)]
+pub enum Address {
+    /// The hostname to serve over TCP.
+    Hostname(String),
+    /// The IP address to serve over TCP.
+    Ip(IpAddr),
+    /// The path to the unix domain socket.
+    #[cfg(unix)]
```

Comment:
Do we need to `cfg` this? I can't see a reason to do so, and it seems more error prone to have it `cfg`d.